Addition of devise function

kawa18sima opened the pull request:
# WHAT
(Sign up, Sign in, Sign out) function added
# WHY
To manage for each account

Commit:
Clone view

I cloned the material.

Commit:
Generate devise view

Using "rails g devise: view" command

Commit:
Replace HTML file

Replaced from generated HTML with clone HTML file

Commit:
Replace css file

Replaced from generated CSS with cloned CSS file

Commit:
migrate change and rake db:migrate

I forgot about migrate so I did.
Since add_index seems to be an error if it is a long character string,
we restricted the length

Commit:
Add permit of name column

In devise, adding a column requires adding a permit, so add a method to
application.controller.rb

Commit:
Add users.scss to load

I imported it so that users.scss can be read.

kawa18sima commented:
前回のプルリクエストの際にmigrateし忘れていたので行いました。
nameカラムをtextにしていたんですが、add_indexには文字制限があるみたいで、長い文字列だとエラーがでました。
その対処として"length: 10"としてカラム値の最初の１０文字を対象にindexを貼っています。
<img width="1440" alt="2017-04-23 2 04 08" src="https://cloud.githubusercontent.com/assets/24731933/25306508/4ae636dc-27c9-11e7-9ebe-6b6d866b12f6.png">
<img width="1440" alt="2017-04-23 2 03 47" src="https://cloud.githubusercontent.com/assets/24731933/25306509/4c42679e-27c9-11e7-938f-3615410aab1e.png">

kinshotomoya commented on app/controllers/application_controller.rb:
```diff
@@ -2,4 +2,13 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
```
ログインしてない時にログインページに飛ぶ実装
サインアウトした時にどこに飛ぶのか実装

上記の二つここに書きましょう！

kinshotomoya reviewed (commented):
見ました！
遅くなりました。

add_indexの件、了解です。
僕も勉強になりました！
ありがとです！

コメントしたところを修正してください！！

Commit:
A description of the path after signing out and authenticate_user!

In application_controller.rb
· Features of the sign in page after signing out
· Addition of processing redirected to sign page when stamping to
another page without signing in

Commit:
Conversion from erb to haml

Changed to haml because the file when devise was generated was erb

kawa18sima commented:
編集しました。
・ログインしてない時にログインページに飛ぶ実装
・サインアウトした時にサインインに飛ぶようにしました

あとdeviseで再生されたファイルがerbだったのでhamlにしました。

kinshotomoya commented:
みました！
いい感じです！！
マージして次行きましょう！！
